torch/csrc/jit/tensorexpr/ir.h

```diff
@@ -538,7 +538,7 @@ class TORCH_API CompareSelect : public ExprNode<CompareSelect> {
       const Expr* ret_val1,
       const Expr* ret_val2,
       CompareSelectOperation cmp_op)
-      : ExprNodeBase(ToDtype<int>()),
+      : ExprNodeBase(ret_val1->dtype()),
```

Comment:
This is great!

Reply:
You probably want a `CHECK` that `ret_val1` and `ret_val2` have the same type

Reply:
done!